From 9555159854ef0283b02fdce73df1333ea5f4c33d Mon Sep 17 00:00:00 2001
From: Purushottam Kushwaha <pkushwah@codeaurora.org>
Date: Thu, 7 Jun 2018 23:21:36 +0530
Subject: [PATCH 1/2] Add wifi_add_or_remove_virtual_intf() to the legacy wifi
 hal

This is used to implement add/remove virtual interfaces for legacy
wifi drivers at runtime.

Change-Id: I5e95fcc350b2e000dab8843d91423235d4310f03
CRs-Fixed: 2257197
Signed-off-by: Satish Patel <tosatishpatel@gmail.com>
---
 include/hardware_legacy/wifi_hal.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/hardware_legacy/wifi_hal.h b/include/hardware_legacy/wifi_hal.h
index 059bb09..b39893f 100644
--- a/include/hardware_legacy/wifi_hal.h
+++ b/include/hardware_legacy/wifi_hal.h
@@ -478,6 +478,9 @@ typedef struct {
                         iface, wifi_radio_mode_change_handler eh);
     wifi_error (*wifi_set_latency_mode)(wifi_interface_handle iface,
                                         wifi_latency_mode mode);
+    wifi_error (*wifi_add_or_remove_virtual_intf)(wifi_interface_handle iface,
+                                                  const char* ifname, u32 iface_type,
+                                                  bool create);
 } wifi_hal_fn;
 wifi_error init_wifi_vendor_hal_func_table(wifi_hal_fn *fn);
 #ifdef __cplusplus
-- 
2.17.1

